samlang-core/src/printer/__tests__/printer-js.test.ts

```diff
@@ -123,5 +123,5 @@ it('HIR expression to JS string test', () => {
         e2: HIR_INT(BigInt(7)),
       })
     )
-  ).toBe('7 != 7');
+  ).toBe('{7 != 7}');
```

Comment:
well, I think it should be `(7) != (7) instead of `{7 != 7}`

Reply:
ohhh you meant `(` and `)`

Reply:
wait I think the binary case should be `(7 != 7)` unless we want `(` and `)` around int literals as well?

Reply:
Good point. I think we only need to add parenthesis around the entire binary expression.

Reply:
And for the `(samlang)[3]` case the parenthesis would be around the `HighIRVariableExpression` and `HighIRNameExpression`?

Reply:
We probably don't need to do anything special with the index expression. It has a very high precedence level in JS, so we don't need parenthesis to disambiguate.